tests/stress_tests/scripts/run_memcheck.py

```diff
@@ -0,0 +1,175 @@
+
+
+def json_load(path_or_string):
+    """ Load json as file or as string
+    """
+    if os.path.isfile(path_or_string):
+        with open(path_or_string, 'r') as json_fp:
+            return json.load(json_fp)
+    else:
+        return json.loads(path_or_string)
+
+
+def main():
+    """Main entry point.
+    """
+
+    # remove additional args (arguments after --)
+    binary_args = []
+    for idx, arg in enumerate(sys.argv):
+        if arg == '--':
+            binary_args = sys.argv[idx+1:]
+            sys.argv = sys.argv[:idx]
+            break
+
+    init_parser = argparse.ArgumentParser(add_help=False)
+    init_parser.add_argument('--timeline_report',
+                             help=f'Create timeline HTML report file name.')
+    init_parser.add_argument('--upload', action="store_true",
+                             help=f'Upload results to database.')
+    init_parser.add_argument('--compare_runs', action="store_true",
+                             help=f'Compare 2 runs.')
+    args = init_parser.parse_known_args()[0]
+
+    parser = argparse.ArgumentParser(
+        description='Run memcheck tests',
+        usage='%(prog)s [options] binary -- [additional args]',
+        parents=[init_parser])
+    parser.add_argument('binary', help='test binary to execute')
+    parser.add_argument('--gtest_parallel', help='Path to gtest-parallel to use.',
+                        default='gtest_parallel')
+    parser.add_argument('-d', '--output_dir',
+                        required=args.timeline_report or args.upload or args.compare_runs,
+                        help='output directory for test logs')
+    parser.add_argument('-w', '--workers', help='number of gtest-parallel workers to spawn')
+
+    parser.add_argument('--db_url',
+                        required=args.timeline_report or args.upload or args.compare_runs,
+                        help='MongoDB URL in a form "mongodb://server:port"')
+    parser.add_argument('--db_collection',
+                        required=args.timeline_report or args.upload or args.compare_runs,
+                        help=f'use collection name in {DATABASE} database',
+                        choices=COLLECTIONS)
+    parser.add_argument('--metadata',
+                        default='{}',
+                        help=f'add extra runtime information, json formated')
+    parser.add_argument('--strip_log_path',
+                        metavar='REMOVE[,REPLACE]',
+                        default='',
+                        help='remove or replace parts of log path')
+    print(' '.join(sys.argv))
+
+    parser.add_argument('--ref_source',
+                        required=args.compare_runs,
+                        help='Source of reference values of MemCheckTests. '
+                             'Should contain path to a folder with logs or '
+                             'JSON-format query to request data from DB.')
+    parser.add_argument('--comparison_out_file',
```

Comment:
Please rename to --comparison_report to match --timeline_report.

Reply:
Done